### PR TITLE
cmd/juju/space: Supercommand and create, 100% tests coverage

### PR DESCRIPTION
**Pull request by dimitern**
Based on dooferlad's branch implementing juju space create, I've
experimented a bit to see how to simplify the tests, make it easier to
write new subcommands and their tests, using a stub API.

### CODE REVIEW

**Comment by jameinel** [comment]
Looks nice to me

File: cmd/juju/space/create.go [hunk 1]
```diff
@@ -4,31 +4,114 @@
...
+		}
+	}
+
+	// Create the new space.
+	err = api.CreateSpace(c.Name, c.CIDRs.SortedValues())
+	if err != nil {
+		return errors.Annotatef(err, "cannot create space %q", c.Name)
+	}
+
+	if c.CIDRs.IsEmpty() {
```
Comment (TheMue) [inline cmd/juju/space/create.go]: Shouldn't this test be before calling `CreateSpace()`?

**Comment by dooferlad** [comment]
Like it!

**Comment by dimitern** [comment]
$$merge$$

**Pull request merged**